src/set.cr

```diff
@@ -152,6 +152,30 @@ struct Set(T)
     set
   end
 
+  # Difference: returns a new set containing elements in this set that are not
+  # present in the other.
+  #
+  #     Set.new([1,2,3,4,5]) - Set.new([2,4])               #=> Set{1, 3, 5}
+  #     Set.new(['a','b','b','z']) - Set.new(['a','b','c']) #=> Set{'z'}
+  def -(other : Set)
+    set = Set(T).new
+    each do |value|
+      set.add value unless other.includes?(value)
+    end
+    set
+  end
```

Comment:
Your example uses an array, @jhass.  Suppose, for the sake of argument, that the other enumerable is a linked list (or, worse, a wrapper around a `C` collection).  It could criss-cross memory for each `yield`.  Yes, it would still be `O(n)`, but not all `O(n)`s are equal :-).

Yes, let others chime in!